.travis/codecoverage.sh

```diff
@@ -1,8 +1,32 @@
 #!/usr/bin/env bash
 
+# fail script immediately on any errors in external commands
+set -e
+
+travis_retry() {
+  local result=0
+  local count=1
+  while [ $count -le 3 ]; do
+    [ $result -ne 0 ] && {
+      echo -e "\n${ANSI_RED}The command \"$@\" failed. Retrying, $count of 3.${ANSI_RESET}\n" >&2
+    }
+    "$@"
+    result=$?
+    [ $result -eq 0 ] && break
+    count=$(($count + 1))
+    sleep 1
+  done
+
+  [ $count -gt 3 ] && {
+    echo -e "\n${ANSI_RED}The command \"$@\" failed 3 times.${ANSI_RESET}\n" >&2
+  }
+
+  return $result
+}
+
 # For now we will use `npm` for globals.
 # Can switch to `yarn` once we no longer support Node.js 0.12:
 # - travis_retry yarn global add coveralls codeclimate-test-reporter
 travis_retry npm install -g coveralls codeclimate-test-reporter
```

Comment:
I wonder if we can try without automatic retrying?  I don't think we have failed `npm install` / `yarn install` invocations very much lately...